ui/packages/consul-ui/app/services/repository/hcp-link.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import RepositoryService from 'consul-ui/services/repository';
+import dataSource from 'consul-ui/decorators/data-source';
+
+export default class HcpLinkService extends RepositoryService {
+  @dataSource('/:partition/:ns/:dc/hcp-link')
+  async fetch({ partition, ns, dc }, { uri }, request) {
+    let result;
+    try {
+      result = (
+        await request`
+      GET /api/hcp/v2/link/global
+    `
+      )((headers, body) => {
+        return {
+          meta: {
+            version: 2,
+            uri: uri,
+          },
+          body: {
+            isLinked: (body.status['consul.io/hcp/link']['conditions'] || []).some(
+              (condition) => condition.type === 'linked' && condition.state === 'STATE_TRUE'
+            ),
+          },
+          headers,
+        };
+      });
+    } catch (e) {
+      // set linked to false if the global link is not found
+      if (e.statusCode === 404) {
+        result = Promise.resolve({ isLinked: false });
+      } else {
+        result = Promise.resolve(null);
```

Comment:
So if we get an error (auth related or anything really), `null` would probably act as an `isLinked: false` as in code if we tried to do a
```hbs
{{#if hcpLlink.isLinked}}
  Say cluster is linked
{{else}}
  <DisplayBanner />
{{/if}}
```

Thinking aloud, is that what we'd want to do?
* If we fail to fetch this result because our ACL token doesn't have enough permissions - this would return `notLinked` - which is okay - since we have code ensuring we only display banner if user has permission to link
* If we fail for other reasons - I think we should probably check that we don't have a definitive result to the `is this cluster linked` in the `hcp-link-status`service, which is what you were saying with `repo return null for the result object (to make it possible to understand when some other system/auth error raised)` I think

So I think this is fine/great! However we hook this up to the `hcp-link-status`, we'll probably have to handle that there!

Reply:
yea, the idea was to add some state, where we could see that `hcpLink` could not actually be observed as linked or not. there are other ways to do it, for sure, like to set {isLinked: null} or so. 

So I open to discuss it